### ISSUE TITLE
Stuck at installation

### ISSUE THREAD

**Issue by zaggynl** [compactd/compactd#34]
I've tried installing it per below
```
ubuntu 16.04
ip by dhcp
fail2ban installed
ssh installed
ran apt update/upgrade/rebooted

**node 8 installation**
nvm is for managing multiple node versions, we need 8 specifically according to compactd manual
curl -o- https://raw.githubusercontent.com/creationix/nvm/v0.33.6/install.sh | bash
restart ssh session
nvm install 8 (managed node versions)

**couchdb installation**
*through repo (leaves it configured, kaput?)*
sudo add-apt-repository ppa:couchdb/stable
sudo apt update
sudo apt install couchdb

*from source*
dependencies for couchdb (json database software)
sudo apt-get --no-install-recommends -y install \
    build-essential pkg-config erlang erlang-reltool \
    libicu-dev libmozjs185-dev libcurl4-openssl-dev \
		python-sphinx

git clone https://github.com/apache/couchdb.git
cd couchdb
./configure
*stupid workaround for stuck at docs* 
(while in ~/couchdb) 
mkdir -p src/docs/build/man/ 
mkdir -p src/docs/build/html/
cd src/docs/build/man/ 
touch apachecouchdb.1 
cd ~/couchdb 
make release
cd rel
sudo mkdir -p /var/lib/couchdb
sudo chown -R couchdb couchdb
sudo chown -R couchdb /var/lib/couchdb
sudo -i -u couchdb /home/zaggynl/couchdb/rel/couchdb/bin/couchdb
switch to new ssh session

**latest ffmpeg installation**
sudo add-apt-repository ppa:mc3man/xerus-media -y  && sudo apt-get update && sudo apt-get install ffmpeg -y && sudo apt-get install frei0r-plugins -y
	
**installing compactd**
sudo apt install make g++
npm install --global compactd
compactd --configure

  Thanks for downloading compactd 1.2.0 !

  This wizard will guide you through the configuration of compactd
  This should only take one or two minutes...

  ✔ FFprobe and FFmpeg found in /usr/bin/ffmpeg
  ✔ You secret is secure (entropy: 1227)
  ✔ Folder /home/zaggynl/.compactd/ is writable
  ✔ CouchDB is available on port 5984

     All good! You can now start configuring compactd

  Now compactd is going to secure you CouchDB collection by setting a password,
  creating the standard collections and adding validation and permissions.

? Do you wish to continue Yes
  ⠐ Configuring database
   Couldn't finish configuration

  ⠠ Configuring database

ssh session with couchdb shows below repeating every 10 seconds or so:
`{database_does_not_exist,[{mem3_shards,load_shards_from_db,"_users",[{file,"src/mem3_shards.erl"},{line,403}]},{mem3_shards,load_shards_from_disk,1,[{file,"src/mem3_shards.erl"},{line,378}]},{mem3_shards,load_shards_from_disk,2,[{file,"src/mem3_shards.erl"},{line,407}]},{mem3_shards,for_docid,3,[{file,"src/mem3_shards.erl"},{line,91}]},{fabric_doc_open,go,3,[{file,"src/fabric_doc_open.erl"},{line,38}]},{chttpd_auth_cache,ensure_auth_ddoc_exists,2,[{file,"src/chttpd_auth_cache.erl"},{line,187}]},{chttpd_auth_cache,listen_for_changes,1,[{file,"src/chttpd_auth_cache.erl"},{line,134}]}]}
`
```

**Comment by vinz243** [comment]
Can you try to run 
```
compactd --configure --verbose
```

**Comment by squaresmile** [comment]
I'm using Ubuntu 17.10, Node 8.9.1 using nvm. I built and installed couchdb 2.2.1 using the guide on the README. I'm also getting the same error about `database_does_not_exist` in couchdb. The configuration gets stuck and keeps retrying so I just killed it.

In the couchdb log, the databases creation was successful but compactd cannot access them.

```
$ compactd --configure --verbose
2017-11-24T04:44:32.406Z           storyboard INFO  ┌── ROOT STORY: Node.js 8.9.1 on Linux 64-bit, SB 3.1.3 [CREATED]
2017-11-24T04:44:32.406Z           storyboard INFO  Log filter: *:DEBUG
2017-11-24T04:44:32.470Z           storyboard INFO  Log filter is now: *:*

  Thanks for downloading compactd 1.2.0 !

  This wizard will guide you through the configuration of compactd
  This should only take one or two minutes...

  ✔ FFprobe and FFmpeg found in /usr/bin/ffmpeg
  ✔ You secret is secure (entropy: 1227)
  ✔ Folder /home/cereal/.compactd/ is writable
  ✔ CouchDB is available on port 5984

     All good! You can now start configuring compactd

  Now compactd is going to secure you CouchDB collection by setting a password,
  creating the standard collections and adding validation and permissions.

? Do you wish to continue Yes
  ⡀ Configuring database
  [object Object]


   Couldn't finish configuration

  ⠠ Configuring database^C

```

**Comment by vinz243** [comment]
Hmm, can youtry running theses commands and run configure again?
```sh
curl -X PUT http://127.0.0.1:5984/_users

curl -X PUT http://127.0.0.1:5984/_replicator

curl -X PUT http://127.0.0.1:5984/_global_changes
```

**Comment by zaggynl** [comment]
@vinz243 

> *   Also, why install make and g++ at the end?
> sorry old note, can be ignored
> * Apt install couchdb should not leave it configured, since the default state is with admin party which is what you want to run the configure command
> * Can you try this tutorial: https://linoxide.com/linux-how-to/install-couchdb-futon-ubuntu-1604/ and stop at $ curl localhost:5984 then run compactd configure

```
reset my snapshot to clean installation and went through below:
sudo add-apt-repository ppa:couchdb/stable
sudo apt-get update
sudo apt-get install couchdb
sudo chown -R couchdb:couchdb /usr/bin/couchdb /etc/couchdb /usr/share/couchdb
sudo chmod -R 0770 /usr/bin/couchdb /etc/couchdb /usr/share/couchdb
sudo systemctl restart couchdb
curl localhost:5984
{"couchdb":"Welcome","uuid":"d26cc289fdaa763280175d97616ab1ae","version":"1.6.1","vendor":{"name":"Ubuntu","version":"16.04"}}
sudo add-apt-repository ppa:mc3man/xerus-media -y  && sudo apt-get update && sudo apt-get install ffmpeg -y && sudo apt-get install frei0r-plugins -y
npm install --global compactd

gyp ERR! configure error
gyp ERR! stack Error: Can't find Python executable "python", you can set the PYTHON env variable.

sudo apt install python
npm install --global compactd

gyp ERR! stack Error: not found: make

sudo apt install make
npm install --global compactd

make: g++: Command not found

sudo apt install g++
npm install --global compactd
compactd --configure --verbose
Please start this script again to configure
compactd --configure --verbose
  ✘ CouchDB is already configured
  Error
   Couldn't finish configuration
2017-11-24T12:11:15.674Z           storyboard INFO  └── ROOT STORY: Node.js 8.9.1 on Linux 64-bit, SB 3.1.3 [CLOSED]

zaggynl@compactd:~$ curl -X PUT http://127.0.0.1:5984/_users
{"error":"file_exists","reason":"The database could not be created, the file already exists."}
zaggynl@compactd:~$ curl -X PUT http://127.0.0.1:5984/_replicator
{"error":"file_exists","reason":"The database could not be created, the file already exists."}
zaggynl@compactd:~$ curl -X PUT http://127.0.0.1:5984/_global_changes
{"error":"illegal_database_name","reason":"Name: '_global_changes'. Only lowercase characters (a-z), digits (0-9), and any of the characters _, $, (, ), +, -, and / are allowed. Must begin with a letter."}
```

**Comment by squaresmile** [comment]
Before configure


```
$ curl -X PUT http://127.0.0.1:5984/_users
{"ok":true}
cereal@ns500530:~$ curl -X PUT http://127.0.0.1:5984/_replicator
{"ok":true}
cereal@ns500530:~$ curl -X PUT http://127.0.0.1:5984/_global_changes
{"ok":true}
```


After configure

````
$ curl -X PUT http://127.0.0.1:5984/_users
{"error":"unauthorized","reason":"You are not a server admin."}
$ curl -X PUT http://127.0.0.1:5984/_replicator
{"error":"unauthorized","reason":"You are not a server admin."}
$ curl -X PUT http://127.0.0.1:5984/_global_changes
{"error":"unauthorized","reason":"You are not a server admin."}
````

**Comment by vinz243** [comment]
@squaresmile did those commands fixed your issue? If not, did it changed the configure script output? the error after the configure are normal btw

@zaggynl the version is outdated, my bad. Can you try:

```
echo "deb https://apache.bintray.com/couchdb-deb xenial main" | sudo tee -a /etc/apt/sources.list
curl -L https://couchdb.apache.org/repo/bintray-pubkey.asc | sudo apt-key add -
sudo apt-get update && sudo apt-get install couchdb
```

then running configure command directly

**Comment by squaresmile** [comment]
Unfortunately, the verbose configuration still has the same error and output. However, I think couchdb threw a different error, saying the user doesn't have the permission to change the document.

````
Nov 24 14:37:37 couchdb[23946]: [error] 2017-11-24T19:37:37.873846Z couchdb@127.0.0.1 <0.2822.0> -------- OS Process Error <0.2888.0> :: {<<"Error">>,{[{<<"stack">>,<<"(\"value\",0,[object Array])@validate_doc_update:1\n([object Object],[object Object],[object Object],null,[object Object],_a,_f,_i)@validate_doc_update:1\n([object Object],null,[object Object],[object Object])@validate_doc_update:1\n(function (d, o, u) {\nfunction _f(t, a) {if (null == t) {throw new TypeError(\"\\\"t\\\" is null or not defined\");}var b = Object(t), c = b.length >>> 0;if (\"function\" != typeof a) {throw new TypeError(\"predicate must be a function\");}for (var d = arguments[1], e = 0; e < c;) {var f = b[e];if (a.call(d, f, e, b)) {return f;}e++;}}\n\n\nfunction _i(r, n, e) {if (null == r) {throw new TypeError(\"\\\"arr\\\" is null or not defined\");}var t = Object(r), a = t.length >>> 0;if (0 === a) {return false;}for (var i = 0 | e, u = Math.max(i >= 0 ? i : a - Math.abs(i), 0); u < a;) {if (t[u] === n) {return true;}u++;}return false;}\n\n\nfunction _a(a, b) {\"use strict\";if (null == a) {throw new TypeError(\"Cannot convert undefined or null to object\");}for (var c = Object(a), d = 1; d < arguments.length; d++) {var e = arguments[d];if (null != e) {for (var f in e) {Object.prototype.hasOwnProperty.call(e, f) && (c[f] = e[f]);}}}return c;}\n\n(function (sc, pe, pr, old, user, assign, find, includes) {var base = [\"_id\", \"_rev\", \"_revisions\", \"_deleted\"];var keys = Object.keys;var doc = assign({}, old || {}, pr || {});var err = find(keys(sc).map(function (k) {return eval(\"(\" + sc[k] + \")\")(k, doc[k]);}), function (el) {return el !== \"\";});if (err) {throw {forbidden: err};}var ex = keys(pr).filter(function (k) {return !includes(base, k);}).filter(function (k) {return !~keys(sc).indexOf(k);});if (ex.length) {throw {forbidden: \"Keys \" + ex.join(\", \") + \" aren't allowed\"};}if (!pe) {return;}if (!old) {var perm = pe.create;for (var i = 0; i < user.roles.length; i++) {var role = user.roles[i];if (perm[role]) {return;}}throw {forbidden: \"User is
Nov 24 14:37:37 couchdb[23946]: not allowed to create documents\"};}var perm = pe.read;var ok = false;for (var i = 0; i < user.roles.length; i++) {var role = user.roles[i];if (perm[role]) {ok = true;}}if (!ok) {throw {forbidden: \"User is not allowed to update documents\"};}for (var i = 0; i < keys(pr).length; i++) {var key = keys(pr)[i];var val = pr[key];if (val !== old[key] && !~pe.upd_fd.indexOf(key) && !includes(base, key)) {throw {forbidden: \"Not allowed to update \" + key};}}}({value: \"function () {\\n return '';\\n }\"}, {read: {_admin: true, end_user: true, user: true}, update: {_admin: true, end_user: true, user: true}, create: {_admin: true, end_user: true, user: true}, upd_fd: [\"value\"]}, d, o, u, _a, _f, _i));},[object Object],[object Array])@./share/server/main.js:1291\n(\"_design/validator\",[object Array],[object Array])@./share/server/main.js:1537\n()@./share/server/main.js:1582\n()@./share/server/main.js:1603\n@./share/server/main.js:1\n">>},{<<"message">>,<<"call to eval() blocked by CSP">>},{<<"fileName">>,<<"validate_doc_update">>},{<<"lineNumber">>,1}]}}
Nov 24 14:37:37 couchdb[23946]: [notice] 2017-11-24T19:37:37.876843Z couchdb@127.0.0.1 <0.2555.0> 4f4cba1a88 127.0.0.1:5984 127.0.0.1 undefined PUT /config/statistician.suggestions.enable 500 ok 468
````

**Comment by vinz243** [comment]
This seems to be a problem with this code https://github.com/compactd/compactd/blob/e3a69ae781aee292c0a05d5a48b7af678bf78210/server/src/features/configure/DatabaseConfigurator.ts#L40

ill try to add more logging and remove those parts which  are not being used anyway and release a new version

**Comment by vinz243** [comment]
@squaresmile can . you try again with version 1.2.1? Use 
```
npm update --global compactd
```

to update and then run script with `--verbose` flag

**Comment by zaggynl** [comment]
@vinz243 
Got a bit further now:
```
curl -o- https://raw.githubusercontent.com/creationix/nvm/v0.33.6/install.sh | bash
restart ssh session
nvm install 8
echo "deb https://apache.bintray.com/couchdb-deb xenial main" | sudo tee -a /etc/apt/sources.list
curl -L https://couchdb.apache.org/repo/bintray-pubkey.asc | sudo apt-key add -
sudo apt-get update && sudo apt-get install couchdb
selected none on couchdb setup question
sudo add-apt-repository ppa:mc3man/xerus-media -y  && sudo apt-get update && sudo apt-get install ffmpeg -y && sudo apt-get install frei0r-plugins -y
sudo apt install make g++
npm install --global compactd
compactd --configure --verbose
2017-11-25T15:10:15.817Z           storyboard INFO  ┌── ROOT STORY: Node.js 8.9.                                                                          1 on Linux 64-bit, SB 3.1.3 [CREATED]
2017-11-25T15:10:15.819Z           storyboard INFO  Log filter: *:DEBUG
2017-11-25T15:10:15.832Z           storyboard INFO  Log filter is now: *:*

  Thanks for downloading compactd 1.2.0 !

  This wizard will guide you through the configuration of compactd
  This should only take one or two minutes...

  ✔ FFprobe and FFmpeg found in /usr/bin/ffmpeg
  ✔ You secret is secure (entropy: 1227)

  Please start this script again to configure

2017-11-25T15:10:18.254Z           storyboard INFO  └── ROOT STORY: Node.js 8.9.                                                                          1 on Linux 64-bit, SB 3.1.3 [CLOSED]
zaggynl@compactd:~$ compactd --configure --verbose
2017-11-25T15:10:21.266Z           storyboard INFO  ┌── ROOT STORY: Node.js 8.9.                                                                          1 on Linux 64-bit, SB 3.1.3 [CREATED]
2017-11-25T15:10:21.267Z           storyboard INFO  Log filter: *:DEBUG
2017-11-25T15:10:21.280Z           storyboard INFO  Log filter is now: *:*

  Thanks for downloading compactd 1.2.0 !

  This wizard will guide you through the configuration of compactd
  This should only take one or two minutes...

  ✔ FFprobe and FFmpeg found in /usr/bin/ffmpeg
  ✔ You secret is secure (entropy: 1227)
  ✔ Folder /home/zaggynl/.compactd/ is writable
  ✔ CouchDB is available on port 5984

     All good! You can now start configuring compactd

  Now compactd is going to secure you CouchDB collection by setting a password,
  creating the standard collections and adding validation and permissions.

? Do you wish to continue Yes
  ⢀ Configuring database
  [object Object]


   Couldn't finish configuration

  ⠈ Configuring database
/var/log/couchdb/couchdb.log shows:
[error] 2017-11-25T15:30:51.105300Z couchdb@127.0.0.1 <0.7914.0> -------- OS Process Error <0.8027.0> :: {<<"Error">>,{[{<<"stack">>,<<"(\"value\",0,[object Array])@validate_doc_update:1\n([object Object],[object Object],[object Object],null,[object Object],_a,_f,_i)@validate_doc_update:1\n([object Object],null,[object Object],[object Object])@validate_doc_update:1\n(function (d, o, u) {\nfunction _f(t, a) {if (null == t) {throw new TypeError(\"\\\"t\\\" is null or not defined\");}var b = Object(t), c = b.length >>> 0;if (\"function\" != typeof a) {throw new TypeError(\"predicate must be a function\");}for (var d = arguments[1], e = 0; e < c;) {var f = b[e];if (a.call(d, f, e, b)) {return f;}e++;}}\n\n\nfunction _i(r, n, e) {if (null == r) {throw new TypeError(\"\\\"arr\\\" is null or not defined\");}var t = Object(r), a = t.length >>> 0;if (0 === a) {return false;}for (var i = 0 | e, u = Math.max(i >= 0 ? i : a - Math.abs(i), 0); u < a;) {if (t[u] === n) {return true;}u++;}return false;}\n\n\nfunction _a(a, b) {\"use strict\";if (null == a) {throw new TypeError(\"Cannot convert undefined or null to object\");}for (var c = Object(a), d = 1; d < arguments.length; d++) {var e = arguments[d];if (null != e) {for (var f in e) {Object.prototype.hasOwnProperty.call(e, f) && (c[f] = e[f]);}}}return c;}\n\n(function (sc, pe, pr, old, user, assign, find, includes) {var base = [\"_id\", \"_rev\", \"_revisions\", \"_deleted\"];var keys = Object.keys;var doc = assign({}, old || {}, pr || {});var err = find(keys(sc).map(function (k) {return eval(\"(\" + sc[k] + \")\")(k, doc[k]);}), function (el) {return el !== \"\";});if (err) {throw {forbidden: err};}var ex = keys(pr).filter(function (k) {return !includes(base, k);}).filter(function (k) {return !~keys(sc).indexOf(k);});if (ex.length) {throw {forbidden: \"Keys \" + ex.join(\", \") + \" aren't allowed\"};}if (!pe) {return;}if (!old) {var perm = pe.create;for (var i = 0; i < user.roles.length; i++) {var role = user.roles[i];if (perm[role]) {return;}}throw {forbidden: \"User is not allowed to create documents\"};}var perm = pe.read;var ok = false;for (var i = 0; i < user.roles.length; i++) {var role = user.roles[i];if (perm[role]) {ok = true;}}if (!ok) {throw {forbidden: \"User is not allowed to update documents\"};}for (var i = 0; i < keys(pr).length; i++) {var key = keys(pr)[i];var val = pr[key];if (val !== old[key] && !~pe.upd_fd.indexOf(key) && !includes(base, key)) {throw {forbidden: \"Not allowed to update \" + key};}}}({value: \"function () {\\n return '';\\n }\"}, {read: {_admin: true, end_user: true, user: true}, update: {_admin: true, end_user: true, user: true}, create: {_admin: true, end_user: true, user: true}, upd_fd: [\"value\"]}, d, o, u, _a, _f, _i));},[object Object],[object Array])@./share/server/main.js:1291\n(\"_design/validator\",[object Array],[object Array])@./share/server/main.js:1537\n()@./share/server/main.js:1582\n()@./share/server/main.js:1603\n@./share/server/main.js:1\n">>},{<<"message">>,<<"call to eval() blocked by CSP">>},{<<"fileName">>,<<"validate_doc_update">>},{<<"lineNumber">>,1}]}}
```

**Comment by vinz243** [comment]
Can you try running :
```

npm install --global compactd@1.2.1
compactd --configure --verbose
```
because you still have version 1.2.0

**Comment by squaresmile** [comment]
The configuration finished successfully with a lot of [warnings](https://gist.github.com/squaresmile/d32f52f80514d235e94f0f989adbccde).

When I ran serve, it said:
````
  Please upgrade your database to the latest version
  You may want to use compactd --upgrade
````
So I ran `compactd --upgrade` and here is the [output](https://gist.github.com/squaresmile/381b84c0fd634a63d16d3b225e075fd1)

However, running `compactd --serve` doesn't start anything. There's no new process running.

**Comment by vinz243** [comment]
Nice, the warnings itself just indicate that some artworks couldn't be found. 

```
 Please upgrade your database to the latest version
  You may want to use compactd --upgrade
```
is a bug tho, need to fix it

anyway you should be able to do `curl  localhost:9000`

**Comment by squaresmile** [comment]
Unfortunately, there's nothing listening on port 9000. `curl localhost:9000` just returned `Connection refused`.  serve doesn't have any error and said `Successfully started compactd in the background`.

**Comment by vinz243** [comment]
Install pm2 using `npm install --global pm2` then run
```
pm2 start --name compactd  $(dirname $(readlink -f $(which compactd)))/server/index.js
```

**Comment by squaresmile** [comment]
Thanks a lot, compactd is running in pm2 now!

However, a different issue arises. When I go to localhost:9000, it says "socket.io is now connected'' and "An error happened during database sync for artists: undefined". In the console of firefox, it says

````
Object { status: 500, name: "indexed_db_went_bad", message: "unknown", error: true, reason: "Failed to open indexedDB, are you in private browsing mode?", result: {…}, stack: "" }
application.js:24:207686
An error happened during database sync for artists: undefined
````

Thank you for the quick response. Should I open a new issue for this one?

**Comment by vinz243** [comment]
This is really weird ... What version of firefox are you running ? What OS ?  Did you change any specific setting? If you have it, can you try in chrome?

**Comment by squaresmile** [comment]
Firefox 57 Windows 10. 

I tried again with Chrome 63 Beta Windows (my stable Chrome 62 is kinda broken at the moment) and I got a different error.  The loading circle got about to 40%:. Here is the log in pm2:

````
0|compactd | 2017-11-25T21:15:14.948Z                 http INFO  GET /database/artworks/artworks%2Flibrary%2Fyahagi-sayuri/large?rev=1-7f5571fdc3d7ffbb35b2ee15ddd0e84b -> http://127.0.0.1:5984/artworks/artworks%2Flibrary%2Fyahagi-sayuri/large?rev=1-7f5571fdc3d7ffbb35b2ee15ddd0e84b
0|compactd | 2017-11-25T21:15:15.009Z                 http INFO  GET /database/artworks/artworks%2Flibrary%2Fyahagi-sayuri/small?rev=1-7f5571fdc3d7ffbb35b2ee15ddd0e84b -> http://127.0.0.1:5984/artworks/artworks%2Flibrary%2Fyahagi-sayuri/small?rev=1-7f5571fdc3d7ffbb35b2ee15ddd0e84b
````

Checking the corresponding file, it has a 299KB 1417x1948 jpeg cover so I guess it caused the issue.

**Comment by vinz243** [comment]
You might want to report the issue with Firefox on PouchDB... can you email me one of the file with this artist so I can try and imrpove the artwork agent ? Also when the circle gets stuck at 40%, do you get any error in the browser console?

**Comment by squaresmile** [comment]
Here are the errors in the Chrome console:
````
Failed to load resource: the server responded with a status of 403 (Forbidden)
i {code: "ETIMEDOUT", status: 0, result: {…}}code: "ETIMEDOUT"result: {ok: false, start_time: Sat Nov 25 2017 16:49:37 GMT-0500 (Eastern Standard Time), docs_read: 0, docs_written: 0, doc_write_failures: 0, …}status: 0__proto__: Error
An error happened during database sync for artworks: ETIMEDOUT
````

[Here ](https://square.ovh/01%20-%20Cherish%20you.flac)is the file with the artist as the only artist. This one has pretty normal 600x600 jpeg cover.

[Here ](https://square.ovh/01%20Xiu!%20%20Empire%20Hana%20Orchestra.flac)is the file with the artist as one of artists in the artist field. This one has a 1417x1948 jpeg (BD size) cover (the cover is also NSFW)

**Comment by vinz243** [comment]
Using the network tab, can you locate the URL which has a 403 error? And the one with the timedout error?

**Comment by squaresmile** [comment]
Console tab ![Console tab](https://i.imgur.com/z3OuN5Y.png)

Network tab ![Network tab](https://i.imgur.com/zlPvclR.png)

**Comment by vinz243** [comment]
Hmm, seems like some database are not being created. I'm gonna create a virtual machine similar to your env and try install it, I think it will be for the better

**Comment by squaresmile** [comment]
Tried
````
$  curl -X PUT http://admin:password@localhost:5984/trackers
{"ok":true}
````
Looks like it's working now :D

**Comment by vinz243** [comment]
Nice, just make sure you padssword is actually something else than password ;)
@zaggynl did you try the fixes here?

**Comment by zaggynl** [comment]
`
nvm install 8 
echo "deb https://apache.bintray.com/couchdb-deb xenial main" | sudo tee -a /etc/apt/sources.list
curl -L https://couchdb.apache.org/repo/bintray-pubkey.asc | sudo apt-key add -
sudo add-apt-repository ppa:mc3man/xerus-media -y
sudo apt-get update && sudo apt-get install couchdb ffmpeg frei0r-plugins make g++ cifs-utils -y
npm install --global compactd@1.2.1 (why does 'npm install compactd' not pull latest by default?)
npm install --global pm2
fstab entry:
//192.168.1.220/dump/MUSIC /mnt/MUSIC cifs guest,uid=zaggynl,gid=users 0 0
sudo mount -a
compactd --configure --verbose
compactd --configure --verbose
set user/password
music path to /mnt/MUSIC
*indexing* (plenty of warnings)
compactd --serve (nothing served as checked per sudo netstat -lp)
compactd --upgrade
Running upgrade...{ error: 'unknown_error',
 reason: 'undefined',
  name: 'unknown_error',
  status: 500,
  message: 'undefined' }
 ⠐ Running upgrade... (taking a long while, 1h+, seeing writes to /var/lib/couchdb/shards/<filenames>)
`

**Comment by vinz243** [comment]
The serve script does not work. Can you try to start
```
node $(dirname $(readlink -f $(which compactd)))/server/index.js
```
and if it runs without error run:

```
pm2 start --name compactd  $(dirname $(readlink -f $(which compactd)))/server/index.js
```

**Comment by zaggynl** [comment]
Appears to work now!
How do I get it to start listening on LAN IP address instead of localhost?

**Comment by vinz243** [comment]
> nano $(dirname $(readlink -f $(which compactd)))/serverr/dist/config.js
> changed 127.0.0.1 to LAN IP address

You  should be editing `~/.compactd/config.json` and add an IP value with either you lan IP adress or `0.0.0.0`. Just wondering how large is your library? Mine has 547 tracks. To count tracks, you can do in your browser console:

```
[...document.getElementsByClassName('album-counter')].reduce((acc, val) => acc + (val.innerHTML.split(' ')[0] -0), 0)
```

And can you eleborate what you mean by slow?

**Comment by zaggynl** [comment]
Just paste:
```
    ip: {
        doc: "The IP address to bind.",
        format: "ipaddress",
        default: "192.168.1.50",
        env: "COMPACTD_IP_ADDRESS",
    },
```
into ~/.compactd/config.json ?

3185 files going by folder properties
`[...document.getElementsByClassName('album-counter')].reduce((acc, val) => acc + (val.innerHTML.split(' ')[0] -0), 0)`
did not return anything

**Comment by vinz243** [comment]
Your config file should just contain a line `"ip": "192.168.1.50"`

Nevermind it doesn't work. i guess i haven't tried yet to optimize for huge libraries, since I dont have one. Can you recreate a quick issue about this problem, since this is no longer linked to the installation? Thanks

**Comment by vinz243** [comment]
For the performance issue, a friend will be giving me around 3000 files so I can try and imrpove performance and I  already know what I have to improve
